teuthology/orchestra/connection.py

```diff
@@ -36,7 +37,12 @@ def create_key(keytype, key):
     elif keytype == 'ssh-ed25519':
         return paramiko.ed25519key.Ed25519Key(data=base64.decodestring(key.encode()))
     else:
-        raise ValueError('keytype must be ssh-rsa, ssh-dss (DSA) or ssh-ed25519')
+        l = "{hostname} {keytype} {key}".format(hostname="x", keytype=keytype, key=key)
```

Comment:
why not use `HostKeyEntry.from_line(l)` to handle all cases? more consistent this way.